articles/other/other-how-delete-sso-user.md

```diff
@@ -0,0 +1,190 @@
+---
+title: How to delete single sign on users | UKCloud Ltd
+description: Describes the process for deleting single sign on (SSO) users
+services: other
+author: Sue Highmoor
+
+toc_rootlink: How To
+toc_sub1: 
+toc_sub2:
+toc_sub3:
+toc_sub4:
+toc_title: Delete SSO users
+toc_fullpath: How To/other-how-delete-sso-user.md
+toc_mdlink: other-how-delete-sso-user.md
+---
+
+# How to delete single sign on users
+
+## Overview
+
+When you use the UKCloud Portal to remove a user from an account that includes single sign on (SSO) enabled services, it's important to note that this does not remove the corresponding SSO user. To remove the SSO user, you must use the Identity & Access Management (IDAM) API.
+
+SSO-enabled services are:
+
+- UKCloud for OpenShift
+
+- UKCloud for OpenStack
+
+For more information about the IDAM API, see [*Understanding UKCloud's Identity & Access Management API*](other-ref-idam.md).
+
+## Deleting a single sign on user
+
+The process for using the IDAM API to delete a single sign on user is as follows:
+
+1. Authenticate with the IDAM API:
+
+    ``` bash
+    token=$(curl -X POST "https://idp.ukcloud.com/auth/realms/client-assured/protocol/openid-connect/token" \
+     -d username="<username>" \
+     -d password="<password>"  \
+     -d grant_type=password    \
+     -d client_id=portal.ukcloud | jq -r '.access_token')
+     ```
+
+    This returns a single sign on token to use for authenticating calls to the IDAM API. For example:
+
+    ``` json
+    {
+     "access_token": "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJUU0F1WERTWUt0SGVEWTI5ZHlPVFZVeHZUMWNhYmUyNUpEVVZzajhVRUhrIn0.eyJqdGkiOiI2YWJmYzkw...",
+     "expires_in": 300,
+     "refresh_expires_in": 1800,
+     "refresh_token": "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJUU0F1WERTWUt0SGVEWTI5ZHlPVFZVeHZUMWNhYmUyNUpEVVZzajhVRUhrIn0.eyJqdGkiOiIyMDkyNjA0...",
+     "token_type": "bearer",
+     "not-before-policy": 0,
+     "session_state": "342ab476-d286-41b9-8b3a-1a236cceefc1"
+    }
+    ```
+
+    For convenience, the token is saved to the `token` variable in your current shell session. Use this variable in the `Authorization: Bearer` header for all further calls to the IDAM API.
+
+2. List users and note the ID of the user you want to delete:
+
+    ``` bash
+    curl -s -H "Authorization: Bearer $token" https://idam.ukcloud.com/v1/users | jq -r '.[] | .id + " " + .username'
```

Comment:
The `jq` filter on this request returns the list of users in the following format:

```
0744f665-2d7a-483c-adb3-72b0effe2356 acc1admin
1d9e52e6-f833-47b9-9cfa-be727510c910 acc1usr1
a089698b-8f32-4a32-ab71-790b015aa662 acc1usr2
```

This should replace the code block on line 69 more cleanly.

Reply:
Fixed